app/src/main/java/com/example/makeitso/screens/login/LoginViewModel.kt

```diff
@@ -62,33 +60,27 @@ class LoginViewModel @Inject constructor(
             return
         }
 
-        viewModelScope.launch(showErrorExceptionHandler) {
-            val oldUserId = accountService.getUserId()
-            accountService.authenticate(email, password) { error ->
-                if (error == null) {
-                    linkWithEmail()
-                    updateUserId(oldUserId, openAndPopUp)
-                } else onError(error)
-            }
+        launchCatching {
+            val oldUserId = accountService.currentUserId
+            accountService.authenticate(email, password)
+            linkWithEmail()
```

Comment:
That was a mistake that I did in the previous version, we shouldn't be trying to link two accounts that already exists. The `linkWithEmail()` method should be called only when creating a new account.

Could you remove the call to the methods `linkWithEmail()` and `updateUserId(...)`? After that, you can also delete these two methods below, and the `val oldUserId`.

Again, that was my mistake, sorry!

Reply:
Done